content/blog/2020-05-googlesheets4-0-2-0/index.Rmarkdown.orig

```diff
@@ -0,0 +1,186 @@
+googlesheets4:::gs4_auth_docs()
+```
+
+We're ecstatic to announce the release of googlesheets4 0.2.0 on CRAN:
+
+[googlesheets4.tidyverse.org](https://googlesheets4.tidyverse.org)
+
+googlesheets4 is a package to work with Google Sheets from R. Although version 0.1.0 debuted on CRAN in late 2019, I've waited to blog about it until I considered googlesheet4 a replacement for an older package called googlesheets. That day is here! In particular, googlesheets 0.2.0 can create and edit Sheets.
+
+Install googlesheets4 from CRAN like so:
+
+```{r eval = FALSE}
+install.packages("googlesheets4")
+```
+
+Then attach it for use via:
+
+```{r}
+library(googlesheets4)
+```
+
+googlesheets4 is already documented through [several articles](https://googlesheets4.tidyverse.org/articles/index.html) and, as always, you can find detailed notes about all changes in the [change log](https://googlesheets4.tidyverse.org/news/index.html).
+
+## googlesheets is dead! Long live googlesheets4!
+
+googlesheets4 is a reboot of an earlier package called googlesheets.
+
+*Why **4**? Why googlesheets**4**? Did I miss googlesheets1 through 3? No. The idea is to name the package after the corresponding version of the Sheets API. In hindsight, the original googlesheets should have been googlesheets**3**.*
+
+googlesheets4 wraps [v4 of the Sheets API](https://developers.google.com/sheets), whereas the original googlesheets wraps v3. The [v3 API](https://developers.google.com/sheets/api/v3) has been deprecated for a long time. Its shut-off has been postponed several times, but its days are clearly numbered. Gradual shutdown of certain endpoints and scopes is already underway, with complete shutdown [expected on September 30, 2020](https://cloud.google.com/blog/products/g-suite/migrate-your-apps-use-latest-sheets-api) at the time of writing. If you use googlesheets, you **must** switch to googlesheets4 and the time is now.
+
+Change is hard, but let's focus on the positive:
+
+* The v4 API has a better design, which makes it easier to wrap, which
+  translates to a better user experience in client packages.
+* I have gotten a lot better at writing R packages. Some of the fiddly parts
+  of the original googlesheets came from the awkward v3 API, but some of it
+  came from me.
+* The gargle package ([gargle.r-lib.org](https://gargle.r-lib.org)) provides
+  infrastructure common to ~250 Google APIs, including Sheets, Drive, Gmail, and
+  BigQuery. By using gargle for auth, we can provide a consistent auth
+  experience across googlesheets4, googledrive, gmailr, and bigrquery.
+* The googledrive package ([googledrive.tidyverse.org](https://googledrive.tidyverse.org))
+  provides a full-featured client for working with files on Google Drive, which
+  allows googlesheets4 to focus on operations specific to spreadsheets. In the
+  original googlesheets package, about half of the code and effort was actually
+  devoted to Drive, not Sheets.
+
+## Read a Sheet
```

Comment:
There's a lot going on in this section. I wonder if it might be better to separate roughly into auth, referring to a sheet, then referring to a range within a sheet?

Reply:
I'm going to leave this alone. I deliberately don't talk much about auth here. I really think people will be able to run this code interactively, just by following their nose. As soon as you even mention auth, it feels like a certain fraction of the folks latch onto that, turn a bunch of knobs they don't need to, and manage to make things harder on themselves.

Reply:
This section looks great now